lib/data/multipleBackendGateway.js

```diff
@@ -73,6 +74,8 @@ const multipleBackendGateway = {
                     dataStoreName: controllingLocationConstraint,
                     dataStoreType: client.clientType,
                     dataStoreVersionId,
+                    dataStoreSize,
```

Comment:
What do you need size and md5 for?

Reply:
Because for ingestion, we'll be uploading fake objects -> 0byte objects but with the correct size and MD5 in the headers. That way are able to read them and they are not just treated as metadata. The MD5 is to make sure we are reading the right thing.